common/types.go

```diff
@@ -168,3 +169,40 @@ type LogConfig struct {
 	FileCount   uint64
 	TimeTracker bool
 }
+
+type Block struct {
+	Id         string
+	StartIndex int64
+	EndIndex   int64
+	Size       int64
+}
+
+// list that holds blocks containing ids and corresponding offsets
+type BlockOffsetList []*Block
+
+func (bol BlockOffsetList) FindBlocksToModify(offset, length int64) (BlockOffsetList, int64, bool) {
+	// size of mod block list
+	size := int64(0)
+	currentBlockOffset := offset
+	var modBlockList BlockOffsetList
+	// TODO: change this to binary search (logn) for better perf
+	for _, blk := range bol {
+		if currentBlockOffset >= blk.StartIndex && currentBlockOffset <= blk.EndIndex && currentBlockOffset <= offset+length {
+			modBlockList = append(modBlockList, blk)
+			size += blk.Size
+			currentBlockOffset = blk.EndIndex
+		}
+	}
+	return modBlockList, size, offset+length >= bol[len(bol)-1].EndIndex
+}
+
+// NewUUID returns a new uuid using RFC 4122 algorithm with the given length.
+func NewUUID(length int64) []byte {
```

Comment:
Just wondering, does the GoSDK already have a method like this that we can borrow?

Reply:
they have one but it doesn't support the length variable, so I decided to just have a version we maintain